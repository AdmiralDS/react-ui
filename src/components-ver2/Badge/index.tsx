import * as React from 'react';
import styled, { css } from 'styled-components';
import { typography } from '#src/components-ver2/Typography';

type Dimension = 'm' | 's';
type Appearance =
  | 'light'
  | 'info'
  | 'warning'
  | 'success'
  | 'error'
  | 'grey'
  | 'dark'
  | 'lightInactive'
  | 'lightDisable'
  | 'white'
  | 'whiteInactive'
  | 'whiteDisable'
  | 'whiteBlue';

const background = css<{ appearance: Appearance }>`
  background: ${({ appearance, theme }) => {
    switch (appearance) {
      case 'info':
        return theme.color['Primary/Primary 60 Main'];
      case 'warning':
        return theme.color['Warning/Warning 50 Main'];
      case 'success':
        return theme.color['Success/Success 50 Main'];
      case 'error':
        return theme.color['Error/Error 60 Main'];
      case 'grey':
        return theme.color['Neutral/Neutral 50'];
      case 'dark':
        return theme.color['Neutral/Neutral 80'];
      case 'whiteBlue':
        return theme.color['Special/Static White'];
      case 'white':
      case 'whiteInactive':
      case 'whiteDisable':
        return theme.color['Neutral/Neutral 00'];
      case 'light':
      case 'lightInactive':
      case 'lightDisable':
      default:
        return theme.color['Neutral/Neutral 10'];
    }
  }};
`;

const color = css<{ appearance: Appearance }>`
  color: ${({ appearance, theme }) => {
    switch (appearance) {
      case 'info':
      case 'warning':
      case 'success':
      case 'error':
      case 'grey':
        return theme.color['Special/Static White'];
      case 'dark':
        return theme.color['Neutral/Neutral 00'];
      case 'whiteBlue':
        return theme.color['Primary/Primary 60 Main'];
      case 'lightInactive':
      case 'whiteInactive':
        return theme.color['Neutral/Neutral 50'];
      case 'whiteDisable':
      case 'lightDisable':
        return theme.color['Neutral/Neutral 30'];
      case 'white':
      case 'light':
      default:
        return theme.color['Neutral/Neutral 90'];
    }
  }};
`;

export const BadgeComponent = styled.div<{ dimension: Dimension; appearance: Appearance }>`
  position: relative;
  box-sizing: border-box;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: ${({ dimension }) => (dimension === 's' ? '0 5px' : '0 7px')};
  height: ${({ dimension }) => (dimension === 's' ? '16px' : '20px')};
  border-radius: ${({ dimension }) => (dimension === 's' ? '8px' : '10px')};
  ${({ dimension }) => (dimension === 's' ? typography['Caption/Caption 1'] : typography['Body/Body 2 Long'])}
  ${background}
  ${color}
  user-select: none;
`;

export interface BadgeProps extends React.HTMLAttributes<HTMLDivElement> {
  /** Размер компонента */
  dimension?: Dimension;
  /** Внешний вид компонента */
  appearance?: Appearance;
}

export const Badge: React.FC<BadgeProps> = ({ children, dimension = 'm', appearance = 'light', ...props }) => {
  const ariaLabel = `Количество ${children}`;
  return (
    <BadgeComponent dimension={dimension} appearance={appearance} aria-label={ariaLabel} {...props}>
      {children}
    </BadgeComponent>
  );
};

Badge.displayName = 'Badge';
