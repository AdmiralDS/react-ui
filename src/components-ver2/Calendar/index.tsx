import React, { useEffect, useState } from 'react';

import { DEFAULT_LOCALE_NAME, DEFAULT_YEAR_COUNT } from './constants';
import { getDefaultDateValidator } from './validator';
import { DayNames, Month, Months, Panel, Years } from './components';
import {
  addMonths,
  addYears,
  after,
  before,
  changeTime,
  equal,
  setMonth,
  setYear,
  subMonths,
  subYears,
  valid,
} from './date-utils';
import { CalendarComponent } from './styled/CalendarComponent';
import { ViewScreenType } from './interfaces';
import type { CalendarPropType, IDateSelectionProps } from './interfaces';

export type { CalendarPropType, IDateSelectionProps, ViewScreenType };

export const Calendar = React.forwardRef<HTMLDivElement, CalendarPropType>(
  (
    {
      startDate,
      endDate,
      validator,
      minDate,
      maxDate,
      selected,
      range,
      tooltipContainer,
      localeName = DEFAULT_LOCALE_NAME,
      filterDate,
      currentActiveView,
      currentActiveViewImportant,
      onChange,
      onDateIncreaseDecrease,
      onMonthSelect,
      onYearSelect,
      onViewEnter,
      onViewLeave,
      onViewMonthSelect,
      onViewYearSelect,
      ...props
    },
    ref,
  ) => {
    const getInitialViewDate = () => {
      const current = new Date();
      if (selected) {
        return selected;
      } else {
        if (minDate && before(current, minDate)) {
          return minDate;
        } else if (maxDate && after(current, maxDate)) {
          return maxDate;
        }
      }
      return current;
    };

    // дата, которую отображаем (в том числе в верхней панели)
    const [viewDate, setViewDate] = useState(getInitialViewDate());
    // активная дата, на которой сейчас ховер
    const [activeDate, setActiveDate] = useState<Date | null>(null);
    // отображаем выбор года
    const [yearsView, setYearsView] = useState(false);
    // отображаем выбор месяца
    const [monthsView, setMonthsView] = useState(false);

    // показать экран выбора года или месяца в зависимости от внешнего параметра currentActiveView
    useEffect(() => {
      if (currentActiveView === 'MONTH') {
        setYearsView(false);
        setMonthsView(true);
      }
      if (currentActiveView === 'YEAR') {
        setMonthsView(false);
        setYearsView(true);
      }
      if (currentActiveView === 'DAY') {
        setYearsView(false);
        setMonthsView(false);
      }
    }, [currentActiveView]);

    useEffect(() => {
      setViewDate(getInitialViewDate());
    }, [selected]);

    useEffect(() => {
      if (range && startDate) {
        changeYear(startDate.getFullYear());
        changeMonth(startDate.getMonth());
      }
    }, []);

    useEffect(() => {
      yearsView ? onViewEnter && onViewEnter('YEAR') : onViewLeave && onViewLeave('YEAR');
    }, [yearsView]);

    useEffect(() => {
      monthsView ? onViewEnter && onViewEnter('MONTH') : onViewLeave && onViewLeave('MONTH');
    }, [monthsView]);

    const getValidator = () => {
      return validator || getDefaultDateValidator(minDate, maxDate);
    };

    const handleYearsViewShow = () => {
      if (!currentActiveViewImportant) {
        setYearsView(true);
        setMonthsView(false);
      }
      onViewYearSelect && onViewYearSelect();
    };
    const handleYearsViewHide = () => {
      if (!currentActiveViewImportant) {
        setYearsView(false);
      }
      onViewYearSelect && onViewYearSelect();
    };

    const handleMonthsViewShow = () => {
      if (!currentActiveViewImportant) {
        setYearsView(false);
        setMonthsView(true);
      }
      onViewMonthSelect && onViewMonthSelect();
    };
    const handleMonthsViewHide = () => {
      if (!currentActiveViewImportant) {
        setMonthsView(false);
      }
      onViewMonthSelect && onViewMonthSelect();
    };

    const changeYear = (year: number) => setViewDate((date) => setYear(date, year));
    const changeMonth = (month: number) => setViewDate((date) => setMonth(date, month));

    const increaseYear = () =>
      setViewDate((date) => {
        const increase = addYears(date, yearsView ? DEFAULT_YEAR_COUNT : 1);
        onDateIncreaseDecrease && onDateIncreaseDecrease(increase);
        return increase;
      });
    const decreaseYear = () =>
      setViewDate((date) => {
        const decrease = subYears(date, yearsView ? DEFAULT_YEAR_COUNT : 1);
        onDateIncreaseDecrease && onDateIncreaseDecrease(decrease);
        return decrease;
      });

    const increaseMonth = () =>
      setViewDate((date) => {
        const increase = addMonths(date, 1);
        onDateIncreaseDecrease && onDateIncreaseDecrease(increase);
        return increase;
      });
    const decreaseMonth = () =>
      setViewDate((date) => {
        const decrease = subMonths(date, 1);
        onDateIncreaseDecrease && onDateIncreaseDecrease(decrease);
        return decrease;
      });

    const handleDayMouseEnter = (day: Date, _: any) => setActiveDate(day);
    const handleMonthMouseLeave = () => setActiveDate(null);

    const handleDayClick = (day: Date, event: any) => {
      let date = day;
      if (range || !equal(selected, date)) {
        date = changeTime(date, selected);
        if (range) {
          if (!startDate && !endDate) {
            onChange([date, null], event);
          } else if (startDate && !endDate) {
            if (before(date, startDate)) {
              onChange([date, null], event);
            } else {
              onChange([startDate, date], event);
            }
          } else if (!startDate && endDate) {
            if (before(date, endDate)) {
              onChange([date, endDate], event);
            } else {
              onChange([date, null], event);
            }
          }
          if (startDate && endDate) {
            onChange([date, null], event);
          }
        } else {
          onChange(date, event);
        }
      }
    };

    const handleYearClick = (date: Date) => {
      changeYear(date.getFullYear());
      !currentActiveViewImportant && setYearsView(false);
      onYearSelect && onYearSelect(date);
    };

    const handleMonthClick = (date: Date) => {
      changeMonth(date.getMonth());
      !currentActiveViewImportant && setMonthsView(false);
      onMonthSelect && onMonthSelect(date);
    };

    const renderPanel = () => (
      <Panel
        viewDate={viewDate}
        minDate={minDate}
        maxDate={maxDate}
        yearsView={yearsView}
        monthsView={monthsView}
        localeName={localeName}
        onYearsViewShow={handleYearsViewShow}
        onYearsViewHide={handleYearsViewHide}
        onMonthsViewShow={handleMonthsViewShow}
        onMonthsViewHide={handleMonthsViewHide}
        onNext={yearsView ? increaseYear : increaseMonth}
        onPrevious={yearsView ? decreaseYear : decreaseMonth}
        tooltipContainer={tooltipContainer}
      />
    );

    const renderMonth = () => (
      <>
        <DayNames date={viewDate} localeName={localeName} />
        <Month
          day={viewDate}
          startDate={startDate}
          endDate={endDate}
          selected={selected}
          activeDate={activeDate}
          range={range}
          localeName={localeName}
          validator={getValidator()}
          filterDate={filterDate}
          onMouseEnter={handleDayMouseEnter}
          onMouseLeave={handleMonthMouseLeave}
          onClick={handleDayClick}
        />
      </>
    );

    const renderMonths = () => (
      <Months
        viewDate={viewDate}
        startDate={startDate}
        endDate={endDate}
        selected={selected}
        localeName={localeName}
        range={range}
        validator={getValidator()}
        onClick={handleMonthClick}
      />
    );

    const renderYears = () => (
      <Years
        viewDate={viewDate}
        startDate={startDate}
        endDate={endDate}
        selected={selected}
        localeName={localeName}
        range={range}
        validator={getValidator()}
        onClick={handleYearClick}
      />
    );
    return (
      <CalendarComponent yearsView={yearsView} monthsView={monthsView} {...props} ref={ref}>
        {renderPanel()}
        {yearsView && renderYears()}
        {monthsView && renderMonths()}
        {!yearsView && !monthsView && renderMonth()}
      </CalendarComponent>
    );
  },
);

Calendar.defaultProps = {
  localeName: DEFAULT_LOCALE_NAME,
};

Calendar.displayName = 'Calendar';
