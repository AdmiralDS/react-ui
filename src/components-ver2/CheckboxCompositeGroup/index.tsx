import React, { Children, cloneElement, FC, HTMLAttributes, isValidElement } from 'react';

import { CompositeWrapper } from './CompositeWrapper';
import type { CheckboxDimension } from '../Checkbox/CheckboxDimension';

export interface CheckboxCompositeGroupProps extends HTMLAttributes<HTMLDivElement> {
  dimension?: CheckboxDimension;
  disabled?: boolean;
}

export const CheckboxCompositeGroup: FC<CheckboxCompositeGroupProps> = ({
  children,
  disabled = false,
  dimension = 'm',
  ...props
}) => {
  const renderCheckboxGroup = () => {
    return Children.map(children, (child: React.ReactNode) => {
      if (!isValidElement(child)) {
        return null;
      }
      return cloneElement(child, {
        disabled: disabled || child.props.disabled,
        dimension: dimension,
        ...child.props,
      });
    });
  };
  return (
    <CompositeWrapper dimension={dimension} disabled={disabled} {...props}>
      {renderCheckboxGroup()}
    </CompositeWrapper>
  );
};

CheckboxCompositeGroup.displayName = 'CheckboxCompositeGroup';
