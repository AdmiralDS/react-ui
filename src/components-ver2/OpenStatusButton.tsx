import { ReactComponent as ChevronDownOutline } from '@admiral-ds/icons/build/system/ChevronDownOutline.svg';
import styled from 'styled-components';

export type Appearance = 'primary' | 'white';

export const OpenStatusButton = styled(ChevronDownOutline)<{ $isOpen?: boolean; appearance?: Appearance }>`
  transition: transform 0.3s ease-in-out;
  transform: rotate(${(p) => (p.$isOpen ? 180 : 0)}deg);
  & *[fill^='#'] {
    fill: ${({ appearance, theme }) =>
      appearance === 'white' ? theme.color['Special/Static White'] : theme.color['Neutral/Neutral 50']};
  }

  &:hover *[fill^='#'] {
    fill: ${(props) => (props.appearance === 'white' ? '' : props.theme.color['Primary/Primary 70'])};
  }

  &[data-disabled] {
    pointer-events: none;

    *[fill^='#'] {
      fill: ${({ theme }) => theme.color['Neutral/Neutral 30']};
    }
  }
`;
