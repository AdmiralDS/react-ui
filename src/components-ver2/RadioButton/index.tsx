import * as React from 'react';

import { Hint, Input, InputContainer, RadioButtonComponent, Span } from '#src/components-ver2/RadioButton/style';

type Dimension = 'm' | 's';

export interface RadioButtonProps extends React.InputHTMLAttributes<HTMLInputElement> {
  /** Состояние кнопки, выбрана или нет */
  checked?: boolean;
  /** Отключение кнопки */
  disabled?: boolean;
  /** Размер радиокнопки */
  dimension?: Dimension;
  /** Дополнительный текст (подсказка), который выводится нижней строкой */
  extraText?: React.ReactNode;
}

export const RadioButton = React.forwardRef<HTMLInputElement, RadioButtonProps>(
  ({ children, disabled, dimension = 'm', extraText, className, ...props }, ref) => {
    return (
      <RadioButtonComponent disabled={disabled} className={className} dimension={dimension}>
        <InputContainer dimension={dimension}>
          <Input ref={ref} type="radio" disabled={disabled} dimension={dimension} {...props} />
          <Span disabled={disabled} dimension={dimension} />
        </InputContainer>
        {children}
        {extraText && (
          <Hint disabled={disabled} dimension={dimension}>
            {extraText}
          </Hint>
        )}
      </RadioButtonComponent>
    );
  },
);

RadioButton.displayName = 'RadioButton';
