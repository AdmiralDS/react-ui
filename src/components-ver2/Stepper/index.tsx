import React, { HTMLAttributes } from 'react';
import type { FC } from 'react';

import StepperContext from '#src/components-ver2/Stepper/StepperContext';
import { List } from '#src/components-ver2/Stepper/style';

export * from '#src/components-ver2/Stepper/StepContent';
export * from '#src/components-ver2/Stepper/Step';

export type Orientation = 'horizontal' | 'vertical';

export interface StepperProps extends HTMLAttributes<HTMLUListElement> {
  /** Ориентация компонента */
  orientation?: Orientation;
  /** Индекс активного шага, нумерация шагов начинается с 0.
   * Этот параметр будет перезаписан, если какому-то шагу задан параметр active равный true
   */
  activeStep: number;
  /** Количество строк в шаге, все шаги по высоте вмещают одинаковое количество строк */
  lineClamp?: 1 | 2 | 3;
  /** Ширина шага */
  stepWidth?: number | string;
  /** В последнем шаге опционально можно выключать статусную полосу */
  hideLastStepLine?: boolean;
}

export const Stepper: FC<StepperProps> = ({
  orientation = 'horizontal',
  activeStep = -1,
  lineClamp = 3,
  stepWidth,
  hideLastStepLine = false,
  children,
  ...props
}) => {
  const listRef = React.useRef<any>();
  const steps = React.Children.toArray(children).map((step, index) => {
    if (!React.isValidElement(step)) {
      return null;
    }
    return activeStep === index
      ? React.cloneElement(step, {
          index,
          role: 'listitem',
          'aria-current': 'step',
          hideLine: index === React.Children.toArray(children).length - 1 && hideLastStepLine,
          ...step.props,
        })
      : React.cloneElement(step, {
          index,
          role: 'listitem',
          hideLine: index === React.Children.toArray(children).length - 1 && hideLastStepLine,
          ...step.props,
        });
  });
  const contextValue = React.useMemo(
    () => ({ activeStep, orientation, lineClamp, stepWidth }),
    [activeStep, orientation, lineClamp, stepWidth],
  );

  React.useEffect(() => {
    if (listRef.current && orientation === 'horizontal') {
      const activeNode = listRef.current.childNodes[activeStep] || listRef.current.firstChild;
      listRef.current.scrollLeft = activeStep === 0 ? activeNode.offsetLeft : activeNode.offsetLeft - 16;
    }
  }, [activeStep, listRef.current, steps]);

  return (
    <StepperContext.Provider value={contextValue}>
      <List ref={listRef} role="list" orientation={orientation} {...props}>
        {steps}
      </List>
    </StepperContext.Provider>
  );
};

Stepper.displayName = 'Stepper';
