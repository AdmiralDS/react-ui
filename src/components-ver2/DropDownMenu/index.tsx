import type { HTMLAttributes, MutableRefObject, ReactNode } from 'react';
import * as React from 'react';
import { ReactElement, useRef } from 'react';
import styled, { css } from 'styled-components';
import { Dropdown } from '#src/components-ver2/Dropdown';
import { keyboardKey } from '#src/components/common/keyboardKey';
import { refSetter } from '#src/components/common/utils/refSetter';

export type DropDownDimensions = 'l' | 'm' | 's';

export const menuListHeights = css<{ dimension: DropDownDimensions }>`
  max-height: ${({ dimension }) => {
    switch (dimension) {
      case 'l':
        return `${48 * 6}px`;
      case 'm':
        return `${40 * 6}px`;
      case 's':
        return `${32 * 6}px`;
      default:
        return `${48 * 6}px`;
    }
  }};
`;

const StyledDropdown = styled(Dropdown)<{ dimension: DropDownDimensions }>`
  padding: 8px 0;
  overflow-x: hidden;
  overflow-y: auto;
  ${menuListHeights};
  cursor: pointer;
  outline: none;
  margin: 0;
`;

const MenuList = styled.ul``;

export interface DropDownMenuProps extends HTMLAttributes<HTMLUListElement> {
  /** Размер Меню */
  dimension?: DropDownDimensions;
  /** Элементы содержимого */
  children: ReactNode;
  /**
   *  Позволяет выравнивать контейнер с компонентами относительно тарджет компонента
   *  https://developer.mozilla.org/en-US/docs/Web/CSS/align-self
   */
  alignSelf?: 'auto' | 'flex-start' | 'flex-end' | 'center' | 'baseline' | 'stretch';
}

export const DropDownMenu = React.forwardRef<HTMLUListElement | null, DropDownMenuProps>(
  ({ children, dimension = 'l', alignSelf, ...props }, ref) => {
    const menuRef: MutableRefObject<HTMLUListElement | null> = useRef(null);
    const [hovered, setHovered] = React.useState<string | undefined>('');

    const handleKeyUp = (e: React.KeyboardEvent) => {
      const code = keyboardKey.getCode(e);
      const activeOption = (e.target as HTMLElement).id;
      switch (code) {
        case keyboardKey[' ']: {
          e.preventDefault();
          break;
        }
        case keyboardKey.ArrowUp: {
          setHovered(activeOption);
          e.preventDefault();
          break;
        }
        case keyboardKey.ArrowDown: {
          setHovered(activeOption);
          e.preventDefault();
          break;
        }
      }
    };

    const renderChildrenDropDown = () => {
      return React.Children.map(children, (child: React.ReactNode) => {
        if (!React.isValidElement(child)) {
          return null;
        }

        const detectedSubGroup = child.props.children?.some?.((element: ReactElement) => Array.isArray(element));

        if (detectedSubGroup) {
          return child.props.children.flat().map((child: ReactElement) => {
            return React.cloneElement(child, {
              hovered: hovered === child.props.id,
              onKeyUp: handleKeyUp,
              ...child.props,
            });
          });
        }

        return React.cloneElement(child, {
          hovered: hovered === child.props.id,
          onKeyUp: handleKeyUp,
          ...child.props,
        });
      });
    };

    return (
      <MenuList ref={refSetter(ref, menuRef)} {...props}>
        <StyledDropdown dimension={dimension} targetRef={menuRef} alignSelf={alignSelf}>
          {renderChildrenDropDown()}
        </StyledDropdown>
      </MenuList>
    );
  },
);

DropDownMenu.displayName = 'DropDownMenu';
