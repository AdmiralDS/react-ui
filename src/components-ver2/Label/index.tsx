import styled from 'styled-components';
import { TYPOGRAPHY } from '#src/components-ver2/Typography';

export const Label = styled.label<{ disabled?: boolean }>`
  text-align: left;
  font-family: ${() => TYPOGRAPHY.fontFamily};
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
  color: ${(props) => props.theme.color['Neutral/Neutral 50']};

  [data-disabled] & {
    color: ${(props) => props.theme.color['Neutral/Neutral 30']};
  }

  [data-focus-within] & {
    color: ${(props) => (props.disabled ? '' : props.theme.color['Neutral/Neutral 50'])};
  }
  [data-required-within] &:before {
    content: '* ';
    color: ${(props) => props.theme.color['Error/Error 60 Main']};
  }
  margin-bottom: 8px;
`;

Label.displayName = 'Label';
