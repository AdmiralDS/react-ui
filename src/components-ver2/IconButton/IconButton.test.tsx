import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { LIGHT_THEME } from '#src/components-ver2/themes';
import { IconButton } from '#src/components-ver2/IconButton';

describe('IconButton', () => {
  test('should render component', () => {
    const { container } = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <IconButton />
      </ThemeProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  test('should focus on component when user on press key tab', () => {
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <IconButton />
      </ThemeProvider>,
    );
    expect(document.body).toHaveFocus();
    userEvent.tab();
    expect(screen.getByRole('button')).toHaveFocus();
  });

  test('should not focus component when user on press key tab', () => {
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <IconButton disabled />
      </ThemeProvider>,
    );
    expect(document.body).toHaveFocus();
    userEvent.tab();
    expect(document.body).toHaveFocus();
  });

  test('should call onClick when user clicks on component', () => {
    const handleClick = jest.fn();
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <IconButton onClick={handleClick} />
      </ThemeProvider>,
    );
    fireEvent.click(screen.getByRole('button'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
