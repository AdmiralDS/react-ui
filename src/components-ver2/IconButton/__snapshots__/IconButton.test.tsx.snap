// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should render component 1`] = `
.c0 {
  box-sizing: border-box;
  display: inline-block;
  border: none;
  border-radius: 4px;
  background: transparent;
  -webkit-tap-highlight-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  vertical-align: center;
}

.c0[data-dimension='xl'] {
  padding: 0;
  height: 56px;
  width: 56px;
}

.c0[data-dimension='l'] {
  height: 48px;
  width: 48px;
}

.c0[data-dimension='m'] {
  height: 40px;
  width: 40px;
}

.c0[data-dimension='s'] {
  height: 32px;
  width: 32px;
}

.c0:hover {
  cursor: pointer;
  background: #F1F2F4;
}

.c0:hover *[fill^='#'] {
  fill: #0A54C2;
}

.c0:active *[fill^='#'] {
  fill: #0949AA;
}

.c0:disabled {
  cursor: not-allowed;
  background-color: transparent;
}

.c0:disabled *[fill^='#'] {
  fill: #A7B0BE;
}

.c1 {
  vertical-align: top;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  overflow: hidden;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 20px;
}

.c1 > * {
  display: inline-block;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  white-space: nowrap;
}

.c1 > *:not(:first-child) {
  margin-left: 8px;
}

.c1 *[fill^='#'] {
  fill: #626F84;
}

.c1 > svg {
  width: 20px;
  height: 20px;
}

<div>
  <button
    class="c0"
    data-dimension="xl"
    type="button"
  >
    <span
      class="c1"
    />
  </button>
</div>
`;
